packages/devtools_app/lib/src/eval_on_dart_library.dart

```diff
@@ -156,25 +158,52 @@ class EvalOnDartLibrary {
     }
   }
 
-  Future<Library> getLibrary(LibraryRef instance, ObjectGroup isAlive) {
+  Future<Library> getLibrary(LibraryRef instance, Disposable isAlive) {
     return getObjHelper(instance, isAlive);
   }
 
-  Future<Class> getClass(ClassRef instance, ObjectGroup isAlive) {
+  Future<Library> getLibraryById(String id, Disposable isAlive) {
+    return getObjByIdHelper(id, isAlive);
+  }
+
+  Future<Class> getClass(ClassRef instance, Disposable isAlive) {
     return getObjHelper(instance, isAlive);
   }
 
-  Future<Func> getFunc(FuncRef instance, ObjectGroup isAlive) {
+  Future<Func> getFunc(FuncRef instance, Disposable isAlive) {
     return getObjHelper(instance, isAlive);
   }
 
   Future<Instance> getInstance(
     FutureOr<InstanceRef> instanceRefFuture,
-    ObjectGroup isAlive,
+    Disposable isAlive,
   ) async {
     return await getObjHelper(await instanceRefFuture, isAlive);
   }
 
+  Future<Instance> getInstanceById(
```

Comment:
These methods should be layered on top of EvalOnDartLibrary rather than embedded directly in EvalOnLibrary. EvalOnLibrary should be for evaluating in the context of any particular library not specific libraries such as widget_inspector or provider_binding.